### ISSUE TITLE
ActivityNotFoundException if no Play Store is installed

### ISSUE THREAD

**Issue by nspo** [TimotheeJeannin/AppRate#3]
`03-20 03:19:49.082: E/AndroidRuntime(6156): android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.VIEW dat=market://details?id=com.something.myapp}`

Problematic line:
`hostActivity.startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + hostActivity.getPackageName())));` in AppRate.java

Solution: Add try/catch block

Importance: Low (is basically only relevant when testing in an emulator I think)

**Comment by nspo** [comment]
https://gist.github.com/Kopfgeldjaeger/5205324
Patch

**Comment by TimotheeJeannin** [comment]
Hi @Kopfgeldjaeger !

Thank you for reporting this issue. 

About a missing Google Play apk, I see two cases : 
- A developer is trying to make AppRate work on an emulator or a phone without Google Play installed.
  In this case I think that the developer should be warned about what's going on. 
  Either by a `log` or a `toast`. 
- A user installed the app without the Google Play. 
  In this case I think that AppRate should not have displayed the rate dialog in the first place.

I think i should:
1. Add a `displayLogs` method that will enable or not logs for AppRate.
2. Before displaying the rate dialog, check if the Google Play apk is installed. 
   - If it is, display the dialog.
   - If it's not, `log` or `toast` something.

What do you think ? 

Tim

**Comment by nspo** [comment]
Hi!
I rewrote some of your code to have more options that I wanted: https://github.com/Kopfgeldjaeger/RateMeMaybe
Another problem was that orientation changes were not handled, so that LogCat output showed WindowLeaked messages. I implemented the dialog with a fragment whose state is saved when the orientation changes.

To get back to this issue: 
I added an option to set whether the main method should be executed if no Play Store is installed (helpful if using the emulator).
See this file:
https://github.com/Kopfgeldjaeger/RateMeMaybe/blob/master/code/src/com/kopfgeldjaeger/ratememaybe/RateMeMaybe.java
Lines 203++, 254++, 336++, 364++

Yours sincerely
